internal/controllers/machine_controller.go

```diff
@@ -98,24 +101,30 @@ func NewMachineReconciler(
 	}
 
 	return &MachineReconciler{
```

Comment:
Can we add at least a simple test?

Reply:
without support of mocking we can just create integration test.

Reply:
we can implement both see my comment: https://github.com/ironcore-dev/libvirt-provider/pull/384#discussion_r1777179923